apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -127,6 +129,11 @@ class PageTemplateModal extends Component {
 		}
 
 		this.setTemplate( slug );
+
+		// Turn off sidebar's rendering of modal
+		if ( this.props.isPromptedFromSidebar ) {
+			this.props.toggleTemplateModal();
```

Comment:
Doesn't the template modal already hide itself on close? I guess I'm curious if there's a reason to avoid using the template modal's own controls for opening/closing itself

Reply:
Oh, yes it does!  I did not when I added that, but apparently the `onRequestClose` prop on the modal handles that for us.

Reply:
Removed that as well as the `isWarningOpen: false` in the setState of the `toggleTemplateModal` function.

Reply:
I misread that at first, so please disregard my previous replies.

Yes the template modal does 'hide' itself, but the component is still mounted and rendering nothing.  When we toggle it this way through our sidebar component, it completely ends the instance of that component.

Reply:
Ok, that makes sense :)